staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity.go

```diff
@@ -287,3 +287,42 @@ type preferredSchedulingTerm struct {
 	nodeSelectorTerm
 	weight int
 }
+
+type RequiredSchedulingTerm struct {
```

Comment:
Name it `RequiredNodeAffinity`

And there is no need to repeat the word "required" in each field. I would name them:

`labelSelector` and `nodeSelector`

Reply:
> Name it `RequiredNodeAffinity`
it also contains nodeSelector part. Does nodeSelector belong to NodeAffinity?

Reply:
`nodeSelector` is also a node affinity. It's just a simplified API which was written earlier (we cannot change its name).